Source/swiftlint-dev/Rules+Register.swift

```diff
@@ -74,53 +75,174 @@ struct NewRuleDetails: Equatable {
+        guard shardSize > 0, !baseRuleNames.isEmpty else {
+            self.totalShards = baseRuleNames.isEmpty ? 0 : 1
+            return
+        }
+        self.totalShards = (baseRuleNames.count + shardSize - 1) / shardSize // Ceiling division
+    }
+}
+
 private extension SwiftLintDev.Rules.Register {
+    /// Number of test classes per shard for optimal parallelization
+    private static let shardSize = 25
+
+    /// Common parent directory of testsDirectory
+    private var testsParentDirectory: URL {
+        testsDirectory.deletingLastPathComponent()
+    }
+
+    /// Generate content for BuiltInRules.swift file
+    private func generateBuiltInRulesFileContent(rulesImportList: String) -> String {
+        """
+        // GENERATED FILE. DO NOT EDIT!
+
+        /// The rule list containing all available rules built into SwiftLint.
+        public let builtInRules: [any Rule.Type] = [
+        \(rulesImportList.indent(by: 4)),
+        ]
+
+        """
+    }
+
+    /// Generate content for Swift test files
+    private func generateSwiftTestFileContent(forTestClasses testClassesString: String) -> String {
+        """
+        // swiftlint:disable file_length file_name single_test_class type_name
+        // swiftlint:disable:previous blanket_disable_command superfluous_disable_command
+        // GENERATED FILE. DO NOT EDIT!
+
+        @testable import SwiftLintBuiltInRules
+        @testable import SwiftLintCore
+        import TestHelpers
+
+        \(testClassesString)
+
+        """
+    }
+
+    /// Generate content for Bazel .bzl files
+    private func generateBzlFileContent(macroInvocations: String) -> String {
+        #"""
+        # GENERATED FILE. DO NOT EDIT!
+
+        load("@build_bazel_rules_swift//swift:swift.bzl", "swift_library", "swift_test")
+
+        def _generated_test_shard(shard_number, copts, strict_concurrency_copts):
+            """Creates a single generated test shard with library and test targets.
+
+            Args:
+                shard_number: The shard number as a string
+                copts: Common compiler options
+                strict_concurrency_copts: Strict concurrency compiler options
+            """
+            swift_library(
+                name = "GeneratedTests_{}.library".format(shard_number),
+                testonly = True,
+                srcs = ["GeneratedTests/GeneratedTests_{}.swift".format(shard_number)],
+                module_name = "GeneratedTests_{}".format(shard_number),
+                package_name = "SwiftLint",
+                deps = [
+                    ":TestHelpers",
+                ],
+                copts = copts + strict_concurrency_copts,
+            )
+
+            swift_test(
+                name = "GeneratedTests_{}".format(shard_number),
+                visibility = ["//visibility:public"],
+                deps = [":GeneratedTests_{}.library".format(shard_number)],
+            )
```

Comment:
I'm not a big fan of having so much static content in a template. As a string it's harder to edit than in a proper editor. So can we have this part in a separate file fixed?